File: ui/task-tracker/src/app/shared/model/postRequests/createOrEditIssuePR.ts
import { IssuePriority } from "../../enums/issue-priority";
import { IssueStatus } from "../../enums/issue-status";
import { IssueType } from "../../enums/issue-type";

export class CreateOrEditIssuePR {
    id: number | undefined;
    name!: string;
    description!: string;
    type!: IssueType;
    status!: IssueStatus;
    priority!: IssuePriority;
    estimate: string | undefined;
    index!: number;
    epicId: number | undefined;
    authorId: number | undefined;
    assigneeId: number | undefined;
    projectId!: number;
}

File: TaskTracker.Core/src/Models/PostRequests/CreateOrEditIssuePR.cs
using TaskTracker.Core.src.Enums;

namespace TaskTracker.Core.src.Models.PostRequests
{
    public class CreateOrEditIssuePR : BasePostRequest
    {
        public int Id { get; set; }
        /// <summary>
        /// Название задачи
        /// </summary>
        public string Name { get; set; } = null!;

        /// <summary>
        /// Описание задачи
        /// </summary>
        public string Description { get; set; } = string.Empty;

        public IssueType Type { get; set; }

        public IssueStatus Status { get; set; }

        public IssuePriority Priority { get; set; }

        /// <summary>
        /// Оценка задачи
        /// </summary>
        public string? Estimate { get; set; } = string.Empty;

        /// <summary>
        /// Порядковый номер задачи внутри проекта
        /// </summary>
        public int Index { get; set; }

        public int? ParentId { get; set; }

        public int? AuthorId { get; set; }

        /// <summary>
        /// Исполнитель
        /// </summary>
        public int? AssigneeId { get; set; }

        public int ProjectId { get; set; }
    }
}
